pinot-common/src/main/java/org/apache/pinot/common/utils/PinotDataType.java

```diff
@@ -586,22 +586,32 @@ public boolean isSingleValue() {
 
   public PinotDataType getSingleValueType() {
     switch (this) {
+      case BYTE:
```

Comment:
If this PR is just adding a check, why does it need to modify the functionality?

Reply:
This change is to get the single value type for single value type. This method should return the same single value type for single value type itself.

Reply:
And this method would be called in https://github.com/apache/incubator-pinot/blob/2cfaed37cf581362b87a36e924cdd5744d430e03/pinot-core/src/main/java/org/apache/pinot/core/data/recordtransformer/DataTypeTransformer.java#L112
If the single value type between source and dest are the same, the data type are the same. E.g. if source is string_array and dest is string, the data type is the same, even though we should mark the flag of `single-value multi-value mismatch`.